File: Sort/QuickSort/QuickSort.py
def quickSort(arr):
    less = list()
    grater = list()
    if(len(arr) <= 1):
        return arr
    else:
        pivot = arr[len(arr) - 1]
        for i in range(len(arr) - 1):
            if(arr[i] < pivot):
                less.append(arr[i])
            else:
                grater.append(arr[i])
        return quickSort(less) + [pivot] + quickSort(grater)
    return arr
if __name__ == "__main__":
    arr = [ 3, 6, 2, 9, 8, 7, 1, 4, 5 ]
    print(quickSort(arr))

File: Sort/QuickSort/QuickSort.java
import java.util.Arrays;
class QuickSort
{
    public static void swap(int[] arr, int a, int b)
    {
        int temp = arr[a];
        arr[a] = arr[b];
        arr[b] = temp;
    }
    
    public static int partition(int[] arr, int left, int right)
    {
        int pivot = right;
        int storeIndex = left;
        
        for(int i=left; i < right; i++)
        {
            if(arr[i] < arr[pivot])
            {
                swap(arr, storeIndex, i);
                storeIndex++;
            }
        }
        swap(arr, storeIndex, pivot);
        return storeIndex;
    }
    
    public static void quickSort (int[] arr, int left, int right)
    {
           if(left < right)
           {
               int pivot = partition(arr, left, right);
               quickSort(arr, left, pivot - 1);
               quickSort(arr, pivot + 1, right);
           }
    }
    
    public static void main(String[] args)
    {
        int[] arr = { 3, 6, 2, 9, 8, 7, 1, 4, 5 };
        quickSort(arr, 0, arr.length - 1);
        System.out.println(Arrays.toString(arr));
    }
}

